lib/clif-backend/src/module_env.rs

```diff
@@ -20,6 +22,8 @@ pub struct ModuleEnv<'module, 'isa> {
     pub signatures: Map<SigIndex, ir::Signature>,
     globals: Map<GlobalIndex, cranelift_wasm::Global>,
     func_bodies: Map<LocalFuncIndex, ir::Function>,
+    pub deduplicated: PrimaryMap<cranelift_wasm::SignatureIndex, SigIndex>,
+    duplicated: HashMap<SigIndex, cranelift_wasm::SignatureIndex>,
```

Comment:
Maybe the naming could be more specific `deduplicated_signatures` `duplicated_signatures` or `signatures_all`/`signatures_unique`

Reply:
This could probably be designed better tbh. I was just trying to get it to work as quickly as possible.